### DISCUSSION TITLE
Sliding window function

### DISCUSSION THREAD

**Discussion by antonkatz** [remeda/remeda#966] [Suggest a Function]
I find myself needing this often:

```
function toWindows(inputArray, size) {
  return Array.from(
    {length: inputArray.length - (size - 1)}, //get the appropriate length
    (_, index) => inputArray.slice(index, index+size) //create the windows
  )
}
```

What does the community think about adding this functionality in?

**Comment by lstkz** [comment]
Does lodash or ramda have a similar function? What's an example use case for it?

**Comment by philihp** [comment]
@lstkz apologies for being 2+ years late here, but perhaps [`aperture`](https://ramdajs.com/docs/#aperture) is what you're looking for?

**Comment by cjquines** [comment]
i think we should do this, it's tc39 stage 1 https://github.com/tc39/proposal-iterator-chunking

**Comment by eranhirsch** [comment]
Isn't this just `chunk`?

**Reply by LeopoldArkham** [reply]
`windows` returns overlapping sub-arrays, unlike `chunk`:
```
const arr = [1, 2, 3, 4, 5, 6];
const w = windows(arr, 3); // [[1, 2, 3], [2, 3, 4], [4, 5, 6]]
```

I know I've used it when manipulating series data (say to process numbers to make a chart) when you want to compute some averages, or in a game-of-life kind of scenario where a cell of data should be affected by some of its neighbours somehow
